File: backend/src/main/java/com/dd25/dietiestates25/dto/request/ListingSearchRequest.java
package com.dd25.dietiestates25.dto.request;

import java.math.BigDecimal;

import com.dd25.dietiestates25.model.enums.ListingType;

public record ListingSearchRequest
(
    String city,

    Double latitude,  
    
    Double longitude,
    
    BigDecimal minPrice,
    
    BigDecimal maxPrice,

    Integer minRooms,

    String energyClass,

    ListingType listingType,

    Boolean nearSchools,

    Boolean nearStops,

    Boolean nearParks
) 
{}

File: frontend/src/app/models/listingSearchRequest.ts
export interface ListingSearchRequest 
{
  city?: string;
  latitude?: number;
  longitude?: number;
  listingType?: 'SALE' | 'RENT' | null;
  minPrice?: number | null;
  maxPrice?: number | null;
  minRooms?: number;
  energyClass?: string;
  nearStops?: boolean;
  nearParks?: boolean;
  nearSchools?: boolean;
  page?: number;
  size?: number;
}